tasks/new_e2e_tests.py

```diff
@@ -205,20 +205,32 @@ def _get_existing_stacks(ctx: Context) -> List[str]:
 
 def _destroy_stack(ctx: Context, stack_name: str):
     # running in temp dir as this is where datadog-agent test
-    # stacks are stored
+    # stacks are stored. It is expected to fail on stacks existing locally
+    # with resources removed by agent-sandbox clean up job
     with ctx.cd(tempfile.gettempdir()):
-        ctx.run(f"pulumi destroy --stack {stack_name} -r --yes --remove --skip-preview", pty=True)
+        ret = ctx.run(
+            f"PULUMI_SKIP_UPDATE_CHECK=true pulumi destroy --stack {stack_name} --yes --remove --skip-preview",
+            pty=True,
+            warn=True,
+        )
+        if ret is not None and ret.exited != 0:
+            # run with refresh on first destroy attempt failure
+            ctx.run(
+                f"PULUMI_SKIP_UPDATE_CHECK=true pulumi destroy --stack {stack_name} -r --yes --remove --skip-preview",
```

Comment:
`-r` here triggeres destroy with refresh. Can change it to `--refresh="true"` to make it more readable